pkg/odo/cli/component/list.go

```diff
@@ -46,28 +57,53 @@ func NewListOptions() *ListOptions {
 
 // Complete completes log args
 func (lo *ListOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	lo.isExperimentalMode = experimental.IsExperimentalModeEnabled()
+	lo.devfilePath = filepath.Join(lo.componentContext, DevfilePath)
+	if lo.isExperimentalMode && util.CheckPathExists(lo.devfilePath) {
+		// Add a disclaimer that we are in *experimental mode*
+		log.Experimental("Experimental mode is enabled, use at your own risk")
+
+		lo.Context = genericclioptions.NewDevfileContext(cmd)
+		lo.Client = genericclioptions.Client(cmd)
+		lo.hasDCSupport, err = lo.Client.IsDeploymentConfigSupported()
```

Comment:
so `hasDCSupport` is related to my convo here https://github.com/openshift/odo/pull/3505#issuecomment-656278199

we're letting s2i components be listed in experimental mode without any flags or context, so why i cant do this in non-experimental mode?

Reply:
That’s in the follow up issue mentioned in the description as the original user story didn’t have this as AC